sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileApiTest.java

```diff
@@ -1958,6 +1960,15 @@ public void appendData() {
         assertTrue(Boolean.parseBoolean(headers.getValue(X_MS_REQUEST_SERVER_ENCRYPTED)));
     }
 
+    @Test
+    public void appendDataInputStreamOptionsError() throws IOException {
+        InputStream mockedData = mock(InputStream.class);
+        doThrow(UncheckedIOException.class).when(mockedData).read();
```

Comment:
Change this to an `IOException` as implementations of `InputStream` should throw `IOException` when reading fails as it's a checked exception in the interface/abstract class.

Also, a recommendation since `InputStream` only requires `read()` to be implemented, just create an anonymous type here instead of using mocking:

```java
InputStream mockedData = new InputStream() {
    @Override
    public int read() throws IOException {
        throw new IOException();
    }
}